Add SDL Wii Remote + Nunchuk profiles

Sam-Belliveau opened the pull request:
# Wii Remote + Nunchuck Controls

Kinda depends on https://github.com/dolphin-emu/dolphin/pull/13961

Basically 

- Nunchuck Stick -> Left Stick
- Nunchuck C / L -> Left Shoulder / Trigger
- Wii Remote DPad -> DPad
- Wii Remote A / B -> A / B or Right Shoulder / Right Bumper
- Wii Remote Shake -> X or Y
- Wii Remote Pointer -> Relative Right Stick, Hide after 5 seconds

Which maps very closely to how you hold the Wii Remote + Nunchuck. I tested it and it works very well in Super Mario Galaxy, New Super Mario Bros, and Rhythm Heaven Fever. The Wii Pointer works very well with relative input + the auto hiding, I had to change the yaw range so that the sensitivity was the same in each axis. 

The IR hides using ```hold(`Right X+` + `Right X-` + `Right Y+` + `Right Y-` < 0.1, 5.0)```

This makes playing games like Super Mario Galaxy much easier. 

I had to change the relative FOV's for the relative IR in order to make the sensitivity feel even in both axis on a 16:9 monitor. The pointer can go off screen, but because the IR resets after 5 seconds it's not a big deal. 

<img width="1195" height="1079" alt="image" src="https://github.com/user-attachments/assets/1199f430-f71b-4cc4-b7de-ed0c32385a3c" />

<img width="1195" height="1079" alt="image" src="https://github.com/user-attachments/assets/279b1668-5c71-4ba4-8e5e-6460f6d7b611" />

<img width="1195" height="1079" alt="image" src="https://github.com/user-attachments/assets/51a47cee-5618-4deb-a942-4d1b2db3c6e5" />

Trihy commented:
Shouldn't be absolute input for pointer on right stick? I mean the pointer will be more aligned to the screen this way. But maybe I'm wrong.

JoshuaVandaele commented on Data/Sys/Profiles/Wiimote/SDL Wiimote + Nunchuk.ini:
```diff
@@ -0,0 +1,35 @@
+IR/Down = `Right Y-`
+IR/Left = `Right X-`
+IR/Right = `Right X+`
+IR/Hide = hold(`Right X+` + `Right X-` + `Right Y+` + `Right Y-` < 0.1, 5.0)
+IR/Recenter = hold(`Right X+` + `Right X-` + `Right Y+` + `Right Y-` < 0.1, 5.0)
```
Is there a reason this is used instead of `IR/Auto-Hide`?

Sam-Belliveau commented:
I was playing Wii sports resort with the Wiimote + Nunchuk + Gyroscope profile after disconnecting the nunchuk.

Sam-Belliveau commented:
But @MayImilae the issue is, if we don't provide something for the user, it doesn't necessarily mean the user will be able to figure it out. 

It took me about an hour of research to make this Wiimote + Nunchuk profile, as I was referencing 3D Allstars, an actual Wii remote + Nunchuk and my Xbox and PS4 controller. 

When you disconnect the Nunchuk the profile becomes an upright Wii remote. And if you think it's useful, I could add a Sideways Wiimote as well. I could also add a classic controller profile. 

I think the advantage to these being stock profiles is that they serve as useful starting points should the user want to edit them. 

Overall I think it's really important to include these and it'll make it easier for users to play their Wii library without necessarily having a Wii remote.

Sam-Belliveau commented:
> Shouldn't be absolute input for pointer on right stick? I mean the pointer will be more aligned to the screen area this way. But maybe I'm wrong. And could be uncomfortable.

I'd recommend trying the profile. I spent a lot of time tuning the IR controls to feel natural, and over all I've found something which made navigating Wii menus feel fun and natural (especially with A and B being on Shoulder and Bumper, which is required to make grabbing Mii's comfortable, among other things)

MayImilae commented:
> Just to be clear there isn't a default here, it's just a profile.

This is what I get for reviewing at 5 in the morning. Well that's good then, but it would be good to include profiles for Horizontal Wii Remote and Classic Controller as well. Both are at least pretty easy.

Sam-Belliveau commented:
@MayImilae 

- "Misc 1" actually does exist on new Xbox controllers. That's actually how I bound it in the first place.

- I added a Gyro Recenter, I made it flicking the right stick in any direction as it was unused otherwise.

- I unbound the Nunchuk Accelerometer and just split the shaking onto X for the Wiimote and Y for the Nunchuk. Apparently in Super Mario Galaxy the different shakes have different purposes.

- I feel like binding d-pad down to anything other than d-pad down in a default configuration might be a little bit confusing. It reminds me of majora's mask bindings, which always confused me. I think if a user really wants to change the binding for d-pad down, its easy enough using this config as a base.

- I fixed the sideways wii remote's configuration. 

- I swapped the classic controllers configuration. 

- I also added a "Touch IR" configuration for PS4 Remotes.

jordan-woyak commented on Data/Sys/Profiles/Wiimote/SDL Sideways Wiimote.ini:
```diff
@@ -0,0 +1,50 @@
+[Profile]
```
I think all the file names should say `Wii Remote` instead of `Wiimote`.
We are trying to use the proper terminology in user-facing strings now.

JoshuaVandaele commented on Data/Sys/Profiles/Wiimote/SDL Sideways Wii Remote.ini:
```diff
@@ -0,0 +1,50 @@
+D-Pad/Down = `Pad S` | `Left Y-`
+D-Pad/Left = `Pad W` | `Left X-`
+D-Pad/Right = `Pad E` | `Left X+`
+IR/Dead Zone = 10.
+IR/Total Pitch = 48.
+IR/Total Yaw = 27.
```
I'm not too familiar with what those settings do, why are we setting them to something instead of letting them grab their default values?

iwubcode replied:
Maybe it's worth adding a comment to mention that?

JoshuaVandaele commented on Data/Sys/Profiles/Wiimote/SDL Classic Controller.ini:
```diff
@@ -0,0 +1,29 @@
+Classic/Left Stick/Down = `Left Y-`
+Classic/Left Stick/Left = `Left X-`
+Classic/Left Stick/Right = `Left X+`
+Classic/Left Stick/Calibration = 100.00
```
Is there a reason we would want to set this to 100 instead of the default? (Idem for all the other `Calibration` values)

jordan-woyak commented on Data/Sys/Profiles/Wiimote/SDL Wii Remote + Nunchuk + GyroIR.ini:
```diff
@@ -0,0 +1,40 @@
+[Profile]
```
I think these file names would maybe be better as `with Gyro Pointing` and `with Touch Pointing` instead of `+ Gyro IR`.
"IR" is an implementation detail, and `with Pointing` matches the `Wii Remote with MotionPlus Pointing` naming.

jordan-woyak commented on Data/Sys/Profiles/Wiimote/SDL Wii Remote + Nunchuk + GyroIR.ini:
```diff
@@ -0,0 +1,40 @@
+Shake/X = `Button X`
+Shake/Y = `Button X`
+Shake/Z = `Button X`
+IMUIR/Recenter = `Right X-` | `Right X+` | `Right Y+` | `Right Y-`
```
`Recenter`ing on pushing the right stick? That seems a bit odd to me.

dreamsyntax commented:
Is this abandoned?

Sam-Belliveau commented:
@dreamsyntax I haven't time as I'm still in university. I think someone could probably pick it up. It's just a few config files so I wouldn't be too upset if it gets closed.
